### PR TITLE
Set connector & connection geometry

### PR DESCRIPTION
**Pull request by robbr48**
### Purpose

- Implement oms3_setConnectorGeometry
- Implement oms3_setConnectionGeometry
- Implement oms3_getConnections 
- Made connections vector in system null-terminated

### Type of Change

- New feature (non-breaking change which adds functionality)

### Checklist

- I have performed a self-review of my own code
- My changes generate no new warnings

### CODE REVIEW

**Comment by robbr48** [comment]
@adeas31 Can you have a look at this? Is it what you need for OMEdit?

File: src/OMSimulatorLib/System.cpp [hunk 1]
```diff
@@ -270,6 +271,21 @@ oms3::Connector *oms3::System::getConnector(const oms3::ComRef &cref)
...
+    oms3::ComRef head = tail.pop_front();
+    auto subsystem = subsystems.find(head);
+    if(subsystem != subsystems.end()) {
+      return subsystem->second->getConnections(tail);
+    }
+  }
+
+  std::vector<oms3::Connection*> temp = connections;
+  //temp.push_back(NULL);
+  return &connections[0];
```
Comment (lochel) [inline src/OMSimulatorLib/System.cpp]: `temp`looks like a relict

File: src/OMSimulatorLib/System.cpp [hunk 2]
```diff
@@ -330,9 +348,65 @@ oms_status_enu_t oms3::System::addConnection(const oms3::ComRef &crefA, const om
        (conB->getCausality() == oms_causality_output && conA->getCausality() != oms_causality_input))
       return logError("Causality mismatch in connection: "+std::string(crefA)+" -> "+std::string(crefB));
 
-    connections.push_back(new oms3::Connection(crefA,crefB));
+    connections[connections.size()-1] = new oms3::Connection(crefA,crefB);
```
Comment (lochel) [inline src/OMSimulatorLib/System.cpp]: Better use `connections.back() = ...`

**Comment by OpenModelica-Hudson** [comment]
The following pull requests are now being [tested](https://test.openmodelica.org/hudson/job/OMSimulator-PR/631//console). It may take a while, depending on the workload.
  - OpenModelica/OMSimulator#366

The changes will be pushed if all tests run successfully.

**Comment by OpenModelica-Hudson** [comment]
**[Linux:](https://test.openmodelica.org/hudson/job/OMSimulator-PR/631/)** The test suite is unstable. :worried:

**Comment by OpenModelica-Hudson** [comment]
The following pull requests are now being [tested](https://test.openmodelica.org/hudson/job/OMSimulator-PR/632//console). It may take a while, depending on the workload.
  - OpenModelica/OMSimulator#366

The changes will be pushed if all tests run successfully.

**Comment by OpenModelica-Hudson** [comment]
All tests were successful. :relaxed:

**Pull request closed without merging**